### PR TITLE
Consistent S3 actions

### PR DESCRIPTION
**Pull request by kiiadi**
Introducing some abstract classes for S3 actions so that we can have consistent behaviour about what to enable when.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/CopyPathAction.kt [hunk 1]
```diff
@@ -1,30 +1,23 @@
...
-            TelemetryService.recordSimpleTelemetry(e.getRequiredData(LangDataKeys.PROJECT), TELEMETRY_NAME, TelemetryResult.Succeeded)
-        }
-    }
+class CopyPathAction(private val project: Project, treeTable: S3TreeTable) : SingleS3ObjectAction(treeTable, message("s3.copy.path"), AllIcons.Actions.Copy) {
 
-    override fun update(e: AnActionEvent) {
-        e.presentation.isEnabled = treeTable.selectedRows.size == 1 && !treeTable.getSelectedNodes().any { it is S3TreeContinuationNode }
+    override fun performAction(node: S3TreeNode) {
+        CopyPasteManager.getInstance().setContents(StringSelection(node.key))
+        TelemetryService.recordSimpleTelemetry(project, TELEMETRY_NAME, TelemetryResult.Succeeded)
```
Comment (kiiadi) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/CopyPathAction.kt]: @hunterwerlla / @rli thoughts on pushing these `TELEMETRY_NAME` constants into a single place where *all* telemetry constants go (for now until we get a generated version) ?

(obviously this would be a separate PR / change).

Comment (hunterwerlla) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/CopyPathAction.kt]: I like that idea

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/CopyPathAction.kt]: Agree
